From 89a7a1ed7ceae9a2e2543a63fcc57885d23ded1a Mon Sep 17 00:00:00 2001
From: Neil Horman <nhorman@redhat.com>
Date: Mon, 25 Aug 2014 15:26:41 -0400
Subject: [redhat] koji: Allow specification of buildid via git notes

Message-id: <1408980401-17772-1-git-send-email-nhorman@redhat.com>
Patchwork-id: 89047
O-Subject: [RHEL7 PATCH] redhat: Allow specification of buildid via git notes
Bugzilla:
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Kyle McMartin <kmcmarti@redhat.com>
RH-Acked-by: Hannes Frederic Sowa <hannes@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>

Allow specification of the buildid via git notes

    Buildid is passed as a note to the HEAD commit being built via rhg-brew in the
    buildid namespace.  That is to say that:
    git notes --ref buildid add -m"<string>" HEAD

    will allow the rhg-brew build target to set <string> as the buildid.

    Note that, in order for this to work, you need to push the notes branch to the
    server as well as the code branch.  This can be automated by adding the
    following to your git configuration:

    [branch "branchname"]
     remote = remotename
     push = +refs/notes/*:refs/notes/*

    Signed-off-by: Neil Horman <nhorman@redhat.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/redhat/koji/Makefile b/redhat/koji/Makefile
index 28e8a8b..3e7830c 100644
--- a/redhat/koji/Makefile
+++ b/redhat/koji/Makefile
@@ -1,7 +1,36 @@
-sources:
+######################################################################
+#This Makefile is here for the sole purpose of building an SRPM
+#within the brew/koji build environment
+#
+#When Brew launches a build with a commit on the end of the brew uri,
+#it runs the BuildSRPMFromSCM task first, which clones the requested
+#git tree, resets to the requested commit, cd's to the directory, and runs
+#'make sources'.  That make targetis responsible for creating an SRPM 
+#that the next brew task will then build.
+#
+#Other targets in this makefile
+#
+#notes: Responsible for pulling any note branches down from the origin
+#
+#buildid: This sets the BUILDID varible, which allows a brew scratch build
+#   To have a unique buildid as set by the developer.
+#   set it with - git notes --ref buildid -m".buildid" HEAD
+######################################################################
+
+notes:
+ git fetch --force origin refs/notes/*:refs/notes/*
+
+buildid: notes
+ $(eval BUILDID := $(shell git notes --ref buildid show HEAD || echo .test))
+
+sources: buildid 
+ echo $(BUILDID) > ../../localversion
  $(MAKE) -C ../ rh-sources
  mv -f ../rpm/SOURCES/* .
  rm -f ./kernel.spec
  mv ../rpm/SPECS/kernel.spec .
 
+test: buildid
+ echo $(BUILDID)
+
 PHONY = sources
-- 
1.7.1